bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java

```diff
@@ -275,21 +294,40 @@ private void updateThingStatus(@Nullable SomfyTahomaState state) {
     public void updateThingChannels(List<SomfyTahomaState> states) {
         Map<String, String> properties = new HashMap<>();
         for (SomfyTahomaState state : states) {
-            logger.trace("processing state: {} with value: {}", state.getName(), state.getValue());
+            logger.trace("{} processing state: {} with value: {}", getURL(), state.getName(), state.getValue());
             properties.put(state.getName(), state.getValue().toString());
-            updateThingChannels(state);
+            if (RSSI_LEVEL_STATE.equals(state.getName())) {
+                // RSSI channel is a dynamic one
+                updateRSSIChannel(state);
+            } else {
+                updateThingChannels(state);
+            }
         }
         updateProperties(properties);
     }
 
+    private void updateRSSIChannel(SomfyTahomaState state) {
+        createRSSIChannel();
+        Channel ch = thing.getChannel(RSSI);
+        if (ch != null && isChannelLinked(ch)) {
```

Comment:
`isChannelLinked` is quite expensive, while `parseTahomaState` seems to be cheap. Is there a special reason to check the channel link state (I know that has been done before, but ...)

Reply:
removed the check for a linked channel